account_dms_field/models/dms_directory.py

```diff
@@ -0,0 +1,30 @@
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+from odoo.addons.dms.tools.file import check_name
+
+
+class DmsDirectory(models.Model):
+    _inherit = "dms.directory"
+
+    @api.constrains("name")
+    def _check_name(self):
+        for record in self:
+            name = record.name
+            if record.res_model == "account.move":
+                name = name.replace("/", "-")
+            if self.env.context.get("check_name", True) and not check_name(name):
+                raise ValidationError(_("The directory name is invalid."))
+            if record.is_root_directory:
+                childs = record.sudo().storage_id.root_directory_ids.name_get()
+            else:
+                childs = record.sudo().parent_id.child_directory_ids.name_get()
+            if list(
+                filter(
+                    lambda child: child[1] == record.name and child[0] != record.id,
+                    childs,
+                )
+            ):
+                raise ValidationError(
+                    _("A directory with the same name already exists.")
+                )
```

Comment:
This is not correct. If any changes need to be done, they should be done in `_prepare_directory_vals()` of `dms.field.template` and change the directory name there (replacing as necessary).

Reply:
@victoralmau Thank you for the review, back at the desk i can fix it

Reply:
@victoralmau please check again